### PR TITLE
Add readme for coredns 

### PR DESCRIPTION
**Pull request by pacoxu**
**What this PR does / why we need it**:
No readme for coredns, only some for kube-dns 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
add some links and descriptions like kube-dns

**Release note**: 

```release-note
NONE
```

### CODE REVIEW

**Comment by MrHohn** [comment]
/assign @rajansandeep 
I think it would be more clear if we keep coredns and kube-dns manifests in separate folders (e.g. dns/coredns, dns/kube-dns), though that requires changing cluster-up scripts.

**Comment by rajansandeep** [comment]
@MrHohn Yes, keeping them in different folders is doable. Is this something we need to prioritize for 1.10 or it can be for later releases?

File: cluster/addons/dns/coredns_README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+## coredns
+
+Users can enable CoreDNS instead of kube-dns for service discovery.
+`coredns` is schedules DNS Pods and Service on the cluster, other pods in cluster
+can use the DNS Service’s IP to resolve DNS names.
+
+* [Administrators guide](https://coredns.io/2018/01/29/deploying-kubernetes-with-coredns-using-kubeadm/)
```
Comment (rajansandeep) [inline cluster/addons/dns/coredns_README.md]: This guide applies to installing CoreDNS via kubeadm and doesn't apply here.

**Comment by MrHohn** [comment]
@rajansandeep No worries, seems fine to do that for later releases.

File: cluster/addons/dns/coredns_README.md (hunk 1 above)
Comment (neolit123) [reply cluster/addons/dns/coredns_README.md]: @rajansandeep i agree.
also given the ongoing transition to CoreDNS the above paragraph would be misleading for the next release. i would Close this PR, until we have settled.

**Comment by neolit123** [comment]
/remove-lifecycle stale

@rajansandeep @MrHohn 

hello, how do you want to proceed with this PR?

@pacoxu
given there are a couple of folders now under `cluster/addons/dns/`
the name of the file should not be `coredns_README.md`, but rather it should go as README.md in here:
https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/dns/coredns

**Comment by neolit123** [comment]
@pacoxu also please sign the CLA.
instructions in the second comment.

**Comment by chrisohaver** [comment]
FYI, `cluster/addons` per the Kubernetes docs has been described as "deprecated" for quite some time now.  

From https://kubernetes.io/docs/concepts/cluster-administration/addons/ :
> There are several other add-ons documented in the deprecated [cluster/addons](https://git.k8s.io/kubernetes/cluster/addons) directory.
Well-maintained ones should be linked to here. PRs welcome!

I believe "here" above refers to the addons docs page itself, not the deprecated [cluster/addons](https://git.k8s.io/kubernetes/cluster/addons) directory.

IMO, if the docs are wrong, they should be fixed.
If they are right, then we should mention that [cluster/addons](https://git.k8s.io/kubernetes/cluster/addons) is deprecated in it's README.  Something like this ...

> This directory is deprecated, please link well maintained add-ons to the  [Addons](https://kubernetes.io/docs/concepts/cluster-administration/addons/) section in the Kubernetes Documentation.

i.e. the reciprocal of the message in the docs.

**Comment by chrisohaver** [comment]
I say close it.  Per my understanding this addons directory is supposed to be deprecated (as of 10 months ago), and is also titled "Legacy Addons".

**Comment by neolit123** [comment]
> I say close it. Per my understanding this addons directory is supposed to be deprecated (as of 10 months ago), and is also titled "Legacy Addons".

ok

/close

**Pull request closed without merging**